pkg/virt-handler/rest/console.go

```diff
@@ -0,0 +1,125 @@
+package rest
+
+import (
+	"github.com/emicklei/go-restful"
+	"github.com/gorilla/websocket"
+	"github.com/libvirt/libvirt-go"
+	"io"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap"
+	"net/http"
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+}
+
+type Console struct {
+	connection virtwrap.Connection
+}
+
+func NewConsoleResource(connection virtwrap.Connection) *Console {
+	return &Console{connection: connection}
+}
+
+func (t *Console) Console(request *restful.Request, response *restful.Response) {
+	console := request.HeaderParameter("console")
+	vm := request.PathParameter("name")
+
+	logging.DefaultLogger().Info().Msgf("Opening connection to console %s on VM %s", console, vm)
+	domain, err := t.connection.LookupDomainByName(vm)
+	if err != nil {
+		if err.(libvirt.Error).Code == libvirt.ERR_NO_DOMAIN {
+			logging.DefaultLogger().Error().Reason(err).Msg("Domain not found.")
```

Comment:
Do we have easy access to the domain to communicate in the logs? (also goes for line 39)